### DISCUSSION TITLE
Eclipse Kura OPC-UA Driver Losing Connection

### DISCUSSION THREAD

**Discussion by gabrielrufino11** [eclipse-kura/kura#5181] [Q&A] [answered]
Hi,

The company that I work for uses Eclipse Kura version 4.1.2. We´re facing some behaviors of the Kura when the PLCs loses connection on the network or the electricity goes down. When that happens, we need to basically update the IP address just to stablish the connection again.

![image](https://github.com/eclipse/kura/assets/19976259/0a76dc8d-da86-430d-b83e-1d6b42da703c)

Is someone facing the same problem?

For exemple, the IP address is 0.0.0.0, then when that happens, we take a zero out, 0.0.0, save, and then put the zero back and save again.

**Comment by nicolatimeus** [comment, marked as answer]
Hi,

Are you using the driver in listen-only mode?

If yes you can try creating a wire graph branch containing a timer connected to an OPC-UA asset with a single channel that periodically reads an arbitrary PLC variable, this should trigger reconnection on connection drop when the timer ticks.

**Reply by gabrielrufino11** [reply]
Thank you so much for the help, I´m gonna try this solution!!

**Reply by mattdibi** [reply]
Thanks @nicolatimeus. @gabrielrufino11 closing this as resolved, feel free to ping us again in case the solution is not working for you.
